filter_tensorflow2-lite-custom: build issue when crosscompiling

Hello all

# Issue Description

version: nnstreamer 2.2.0 / main

We experience build issue in cross compiling nnstreamer for Kirkstone Yocto distribution using some Ubuntu 20.04 LTS build machines.
Cross-compilation environment is x86 build machine and aarch64 host architecture.
Build breaks during link of libnnstreamer_filter_tensorflow2-lite-custom.so library.

`
[227/270] aarch64-poky-linux-g++ [...] -o ext/nnstreamer/tensor_filter/libnnstreamer_filter_tensorflow2-lite-custom.so [...] /.../tensor_filter_tensorflow_lite.cc.o [...]  -ltensorflow2-lite-custom [...]
FAILED: ext/nnstreamer/tensor_filter/libnnstreamer_filter_tensorflow2-lite-custom.so
collect2: error: ld returned 1 exit status
`

Reason is libtensorflow2-lite-custom.so is present in sysroot to be linked - however this lib has x86 arch and can not be linked with (aarch64) tensorflow_lite.cc.o object. Thus libnnstreamer_filter_tensorflow2-lite-custom.so link fails.

Problem is tflite2-custom-support feature support is auto. Hence by default meson.build attempts creating libtensorflow2-lite-custom.so:
https://github.com/nnstreamer/nnstreamer/blob/688906944f8250683a9f761896b91c0d2ceed74f/ext/nnstreamer/tensor_filter/meson.build#L288:
host (aarch64 device) library libtensorflow2-lite-custom.so is created by copying shared lib or linking static lib from x86 build machine:
- shared lib: https://github.com/nnstreamer/nnstreamer/blob/688906944f8250683a9f761896b91c0d2ceed74f/ext/nnstreamer/tensor_filter/meson.build#L297
- static lib: https://github.com/nnstreamer/nnstreamer/blob/688906944f8250683a9f761896b91c0d2ceed74f/ext/nnstreamer/tensor_filter/meson.build#L322

Subsequent link of this x86 lib with an aarch64 objects fails.
Current meson implementation for the filter_tensorflow2-lite-custom feature seems to correct only for builds which are not cross compiled. Cross compiled build will complete only if the build machine does not have above tensorflow static/dynamic libraries.

How to Reproduce
===============
1. Use build machine with one of above listed tensorflow library installed in file system, for instance /usr/lib/libtensorflow2-lite.so
2. Build nnstreamer packages using cross-compilation

Thanks

--- a/ext/nnstreamer/tensor_filter/meson.build
+++ b/ext/nnstreamer/tensor_filter/meson.build
@@ -285,7 +285,7 @@ if tflite2_support_is_available
   )
 
 
-  if get_option('tflite2-custom-support').enabled() or get_option('tflite2-custom-support').auto()
+  if (get_option('tflite2-custom-support').enabled() or get_option('tflite2-custom-support').auto()) and not meson.is_cross_build()
     ## Create libtensorflow2-lite-custom.so support.
 
     ## @todo This assumes the same tflite version with the above.

